constellations/gemini-admin/src/announce/announce.js

```diff
@@ -20,11 +20,51 @@ export class AnnouncePage extends React.Component {
         });
     }
 
+    onChangeCheckbox = (e) => {
+        let successCallback = () => console.log("api success");
+        let failCallback = (err) => {
+            alert("Could not save announcement: " + err.response.data);
+            console.log(err.response);
+        };
+
+        let previous = this.state.list.findIndex(
+            (announcement) => announcement.onHomePage
+        );
+        let current = this.state.list.findIndex(
+            (announcement) => announcement.id == e.target.id
+        );
+
+        this.state.list[current].onHomePage = true;
+        if (previous >= 0) {
+            this.state.list[previous].onHomePage = false;
+        }
+
+        this.setState(this.state);
```

Comment:
this statement here is a bit odd. It seems like you are setting state to be what it already is. But I Also see why you need a statement like this because we are dealing with nested objects here (a list of objects and only one property of that field needs changing).

Here's what I recommend and to make this more "React-friendly".
```javascript
let list = this.state.list;

...change some stuff, edit variable list (NOT this.state.list)...
// maybe even this:
const finalList = ...;

this.setState({list: finalList});
```

It's not good practice to modify `this.state.list` because it gives the impression that values in it can change. Instead, save it into a new variable, modify that, and then replace the old one using `this.setState()`.

Hope that makes sense!